new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItDomainInPV.java

```diff
@@ -0,0 +1,780 @@
+                                .mountPath("/shared"))))) // mounted under /shared inside pod
+                    .volumes(Arrays.asList(
+                        new V1Volume()
+                            .name(pvName)
+                            .persistentVolumeClaim(
+                                new V1PersistentVolumeClaimVolumeSource()
+                                    .claimName(pvcName)),
+                        new V1Volume()
+                            .name("create-weblogic-domain-job-cm-volume")
+                            .configMap(
+                                new V1ConfigMapVolumeSource()
+                                    .name(domainScriptCM)))) //config map containing domain scripts
+                    .imagePullSecrets(isUseSecret ? Arrays.asList(
+                        new V1LocalObjectReference()
+                            .name(OCR_SECRET_NAME))
+                        : null))));
+    String jobName = assertDoesNotThrow(()
+        -> createNamespacedJob(jobBody), "Failed to create Job");
+
+    logger.info("Checking if the domain creation job {0} completed in namespace {1}",
+        jobName, namespace);
+    withStandardRetryPolicy
+        .conditionEvaluationListener(
+            condition -> logger.info("Waiting for job {0} to be completed in namespace {1} "
+                + "(elapsed time {2} ms, remaining time {3} ms)",
+                jobName,
+                namespace,
+                condition.getElapsedTimeInMS(),
+                condition.getRemainingTimeInMS()))
+        .until(jobCompleted(jobName, null, namespace));
+
+    // check job status and fail test if the job failed to create domain
+    V1Job job = getJob(jobName, namespace);
+    if (job != null) {
+      V1JobCondition jobCondition = job.getStatus().getConditions().stream().filter(
+          v1JobCondition -> "Failed".equalsIgnoreCase(v1JobCondition.getType()))
+          .findAny()
+          .orElse(null);
+      if (jobCondition != null) {
+        logger.severe("Job {0} failed to create domain", jobName);
+        List<V1Pod> pods = listPods(namespace, "job-name=" + jobName).getItems();
+        if (!pods.isEmpty()) {
+          logger.severe(getPodLog(pods.get(0).getMetadata().getName(), namespace));
+          fail("Domain create job failed");
+        }
+      }
+    }
+
+  }
+
+  /**
+   * Create a persistent volume.
+   *
+   * @param pvName name of the persistent volume to create
+   * @param domainUid domain UID
+   * @throws IOException when creating pv path fails
+   */
+  private void createPV(String pvName, String domainUid) throws IOException {
+    logger.info("creating persistent volume");
+
+    Path pvHostPath = Files.createDirectories(Paths.get(
+        PV_ROOT, this.getClass().getSimpleName(), pvName));
+    logger.info("Creating PV directory host path {0}", pvHostPath);
+    FileUtils.deleteDirectory(pvHostPath.toFile());
```

Comment:
Can we catch the IOException here ad do an assert instated of throwing IOException

Reply:
Put the code in try block and failing when exception is caught